src/core/manager.c

```diff
@@ -1663,7 +1663,11 @@ static void manager_invoke_notify_message(Manager *m, Unit *u, pid_t pid, const
         assert(m);
         assert(u);
         assert(buf);
-        assert(n > 0);
+
+        if(n <= 0) {
+            log_unit_debug(u, "Got a zero-length notification message. Ignoring");
+            return;
+        }
```

Comment:
If somebody adds another call site, they have to look at the function and see what the prerequisites for the arguments are. assert serves as the documentation for that.